apps/ejabberd/src/mod_blocking.erl

```diff
@@ -0,0 +1,206 @@
+%%%-------------------------------------------------------------------
+%%% @author bartek
+%%% @copyright (C) 2016, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 16. May 2016 18:17
+%%%-------------------------------------------------------------------
+-module(mod_blocking).
+
+-xep([{xep, 191}, {version, "1.2"}]).
+-behaviour(gen_mod).
+-export([start/2,
+         process_iq_get/6,
+         process_iq_set/5,
+         stop/1
+        ]).
+
+-include("ejabberd.hrl").
+-include("jlib.hrl").
+-include("mod_privacy.hrl").
+-type listitem() :: #listitem{}.
+-define(BACKEND, mod_privacy_backend).
+%% API
+-export([]).
+
+
+start(Host, _Opts) ->
+    mod_disco:register_feature(Host, ?NS_BLOCKING),
+    ejabberd_hooks:add(privacy_iq_get, Host,
+        ?MODULE, process_iq_get, 50),
+    ejabberd_hooks:add(privacy_iq_set, Host,
+        ?MODULE, process_iq_set, 50),
+    ok.
+
+stop(Host) ->
+    ejabberd_hooks:delete(privacy_iq_get, Host,
+        ?MODULE, process_iq_get, 50),
+    ejabberd_hooks:delete(privacy_iq_set, Host,
+        ?MODULE, process_iq_set, 50),
+    ok.
+
+process_iq_get(_, ?NS_BLOCKING, _From = #jid{luser = LUser, lserver = LServer}, _, _, _) ->
+    Res = case ?BACKEND:get_privacy_list(LUser, LServer, <<"blocking">>) of
+              {error, not_found} ->
+                  {ok, []};
+              {ok, L} ->
+                  {ok, L};
+              _ ->
+                  {error, e}
```

Comment:
I would catch some Variable instead of wildcard and return it in `{error, Other}` tuple. I think  just`e` is too little meaningful